domains/SignIn.js

```diff
@@ -52,22 +101,118 @@ export default function SignUp({ navigation }) {
               placeholder="johndoe@example.com"
               autoFocus
             />
-            <FormInput
-              label="Password"
-              formikProps={formikProps}
-              formikKey="password"
-              placeholder="Password here"
-              secureTextEntry
-            />
+            {!showPassword ? (
+              <FormInput
+                label="Password"
+                formikProps={formikProps}
+                formikKey="password"
+                placeholder="Password here"
+                secureTextEntry
+              />
+            ) : (
+              <FormInput
+                label="Password"
+                formikProps={formikProps}
+                formikKey="password"
+                placeholder="Password here"
+              />
+            )}
+            <View style={styles.container}>
+              <CheckBox
+                onClick={() => setShowPassword(!showPassword)}
+                isChecked={showPassword}
+                rightText="Show Password"
+              />
+            </View>
             {formikProps.isSubmitting ? (
               <ActivityIndicator />
             ) : (
               <Button title="Submit" onPress={formikProps.handleSubmit} />
             )}
             <Button title="Don't have an account, Sign Up!" onPress={handleSignUp} />
+            <Modal
```

Comment:
This is similar to the comment I made in the other pull request but with a twist. The modal in sign-up can (and will) be used in multiple app screens but this modal will not. I'd create a separate **modal component within this folder** and stick this component there!

Reply:
resolved